include/evse_security/evse_security.hpp

```diff
@@ -86,10 +87,10 @@ class EvseSecurity {
     /// @brief Verifies the given \p certificate_chain for the given \p certificate_type against the respective CA
     /// certificates for the leaf.
     /// @param certificate_chain PEM formatted certificate or certificate chain
-    /// @param certificate_type type of the leaf certificate
+    /// @param certificate_type type of the root certificate for which the chain is verified
     /// @return result of the operation
-    InstallCertificateResult verify_certificate(const std::string& certificate_chain,
-                                                const LeafCertificateType certificate_type);
+    CertificateValidationError verify_certificate(const std::string& certificate_chain,
```

Comment:
Agree, but `CertificateValidationError` was an existing type. @AssemblyJohn your opinion?

Reply:
I think we should stick with something existent. However, that is true, a NoError can be a strange name, maybe change it to 'Valid' instead of 'NoError'?

Reply:
I'm not trying to add an extra type, I suggest to rename the current type `CertificateValidationError` to `CertificateValidationResult` and then change `NoError` to a more logical name.